src/components/StatsPanel/StatsPanel.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+import { useStatsPanel } from './StatsPanel.hook';
+import {
+  CardContainer,
+  StatItem,
+  StatIcon,
+  StatText,
+  StatValue,
+  IconWrapper,
+} from './StatsPanel.styles';
+
+import type { StatsPanelProps } from './StatsPanel.types';
+
+export const StatsPanel = ({ totalTests, passed, failed, waiting }: StatsPanelProps) => {
+  const stats = useStatsPanel({ totalTests, passed, failed, waiting });
+
+  return (
+    <CardContainer>
+      {stats.map((stat) => {
+        const Icon = stat.icon;
+
+        return (
+          <StatItem key={stat.id}>
+            <IconWrapper color={stat.color}>
+              <StatIcon color={stat.color}>
```

Comment:
You pass `color={stat.color}` to <StatIcon />. Since StatIcon is a styled div that currently accepts the prop by generic only, this may forward the prop to the DOM and cause warnings. Use a transient prop name (e.g. $color) and align styled definition accordingly.

**Sugerencia:**
```typescript
Change to <StatIcon $color={stat.color}> and adjust StatIcon's typing to accept $color and use it inside the style callback.
```